libraries/AP_HAL_SITL/SITL_State.cpp

```diff
@@ -688,7 +688,27 @@ void SITL_State::_simulator_servos(struct sitl_input &input)
     }
 
     if (_vehicle == ArduPlane) {
-        throttle = constrain_float((input.servos[2] - 1000) / 1000.0f, 0.0f, 1.0f);
+        float forward_throttle = constrain_float((input.servos[2] - 1000) / 1000.0f, 0.0f, 1.0f);
+        // do a little quadplane dance
+        float hover_throttle = 0.0f;
+        uint8_t running_motors = 0;
+        for (i=0; i<4; i++) {
+            float motor_throttle = constrain_float((input.servos[i+4] - 1000) / 1000.0f, 0.0f, 1.0f);
+            // update motor_on flag
+            if (!is_zero(motor_throttle)) {
+                hover_throttle += motor_throttle;
+                running_motors++;
+            }
+        }
+        if (running_motors > 0) {
+            hover_throttle /= running_motors;
+        }
+        if (!is_zero(forward_throttle)) {
+            throttle = forward_throttle;
+        }
+        else {
```

Comment:
formatting issue here :-)  else on previous line with closing brace for the if-clause...

Reply:
Ok!